pkg/server/server.go

```diff
@@ -842,9 +845,14 @@ func (s *Server) Close() {
 		s.listUsersDispatchThrottler.Close()
 	}
 
+	if s.checkIteratorCacheWaitGroup != nil {
```

Comment:
Nit: you shouldn't need the nil check, because you are always creating one.

Reply:
[0f6ce0f](https://github.com/openfga/openfga/pull/2145/commits/0f6ce0f79506e73a5cf2e4f0dacad633a452a7a9)